Not allowable C function for Go

Example of C code:
```c
#include <stdio.h>
void _(){
        printf("Hello");
}
int main(){
        _();
        return 0;
}
```
Result after transpilation:
```go
func _() {
        noarch.Printf([]byte("Hello\x00"))
}
func main() {
        __init()
        _()
        return
}
func __init() {
}
```
Error:
```
$ go run u.go
# command-line-arguments
./u.go:115:3: cannot use _ as value
```

--- a/tests/function.c
+++ b/tests/function.c
@@ -24,9 +24,13 @@ int mul(int a, int b) {
         return a * b;
 }
 
+int _(){
+	return 42;
+}
+
 int main()
 {
-    plan(11);
+    plan(12);
 
     pass("%s", "Main function.");
 
@@ -55,6 +59,8 @@ int main()
 	f = mul;
 	int j = f(3,4);
 	is_eq(j,12);
+	
+	is_eq(_(),42);
 
     done_testing();
 }

--- a/transpiler/call.go
+++ b/transpiler/call.go
@@ -68,6 +68,7 @@ func transpileCallExpr(n *ast.CallExpr, p *program.Program) (
 	if err != nil {
 		return nil, "", nil, nil, err
 	}
+	functionName = util.ConvertFunctionNameFromCtoGo(functionName)
 
 	// Get the function definition from it's name. The case where it is not
 	// defined is handled below (we haven't seen the prototype yet).

--- a/transpiler/functions.go
+++ b/transpiler/functions.go
@@ -55,6 +55,8 @@ func transpileFunctionDecl(n *ast.FunctionDecl, p *program.Program) (decls []goa
 		p.Function = nil
 	}()
 
+	n.Name = util.ConvertFunctionNameFromCtoGo(n.Name)
+
 	// Always register the new function. Only from this point onwards will
 	// we be allowed to refer to the function.
 	if program.GetFunctionDefinition(n.Name) == nil {

--- a/util/goast.go
+++ b/util/goast.go
@@ -301,7 +301,7 @@ func IsGoKeyword(w string) bool {
 	case "break", "default", "func", "interface", "select", "case", "defer",
 		"go", "map", "struct", "chan", "else", "goto", "package", "switch",
 		"const", "fallthrough", "if", "range", "type", "continue", "for",
-		"import", "return", "var":
+		"import", "return", "var", "_":
 		return true
 	}
 

--- a/util/goast.go
+++ b/util/goast.go
@@ -301,11 +301,19 @@ func IsGoKeyword(w string) bool {
 	case "break", "default", "func", "interface", "select", "case", "defer",
 		"go", "map", "struct", "chan", "else", "goto", "package", "switch",
 		"const", "fallthrough", "if", "range", "type", "continue", "for",
-		"import", "return", "var", "_":
+		"import", "return", "var":
 		return true
 	}
 
 	return false
+}
+
+// ConvertFunctionNameFromCtoGo - convert function name fromC to Go
+func ConvertFunctionNameFromCtoGo(name string) string {
+	if name == "_" {
+		return "__"
+	}
+	return name
 }
 
 func CreateSliceFromReference(goType string, expr goast.Expr) *goast.SliceExpr {

